### PR TITLE
ISPN-14650 RESP GETRANGE

### PR DESCRIPTION
**Pull request by rigazilla**
continuation for
https://issues.redhat.com/browse/ISPN-14650

based on #10931

### CODE REVIEW

File: server/resp/src/main/java/org/infinispan/server/resp/commands/string/GETRANGE.java [hunk 1]
```diff
@@ -30,16 +36,35 @@ public GETRANGE() {
...
+      if (arguments.size() != this.getArity() - 1) {
+         ByteBufferUtils.stringToByteBuf("-ERR wrong number of arguments for 'getrange' command\r\n",
+               handler.allocatorToUse());
+         return handler.myStage();
+      }
       byte[] keyBytes = arguments.get(0);
       int beginIndex = Integer.parseInt(new String(arguments.get(1), StandardCharsets.US_ASCII));
-      int lastIndex = Integer.parseInt(new String(arguments.get(2), StandardCharsets.US_ASCII)) + 1;
-      // TODO deal with negatifs
+      int lastIndex = Integer.parseInt(new String(arguments.get(2), StandardCharsets.US_ASCII));
```
Comment (jabolina) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/string/GETRANGE.java:48]: Can use the `ArgumentUtils.toInt`.

File: server/resp/src/main/java/org/infinispan/server/resp/commands/string/GETRANGE.java [hunk 2]
```diff
@@ -30,16 +36,35 @@ public GETRANGE() {
...
+         begin = Math.max(0, arr.length + begin);
+      }
+      if (end < 0) {
+         end = arr.length + end;
+      } else {
+         end = Math.min(arr.length, end);
+      }
+      // Quick return for oo range
+      if (begin >= end || begin >= arr.length || end <= 0) {
+         return new byte[0];
```
Comment (jabolina) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/string/GETRANGE.java:66]: Maybe create a static empty byte array.

Comment (rigazilla) [reply server/resp/src/main/java/org/infinispan/server/resp/commands/string/GETRANGE.java]: applied both suggestions

**Comment by wburns** [comment]
Integrated into main, thanks @rigazilla !

**Pull request merged**